src/XrdHttp/XrdHttpReq.cc

```diff
@@ -1032,6 +1050,42 @@ void XrdHttpReq::mapXrdErrorToHttpStatus() {
   }
 }
 
+/**
+ * Select the checksum to be computed depending on the userDigest passed in parameter
+ * @param userDigest the digest request from the user (extracted from the Want-Digest header)
+ * @param selectedChecksum the checksum that will be performed
+ */
+void XrdHttpReq::selectChecksum(const std::string &userDigest, std::string & selectedChecksum) {
+    char * configChecksumList;
+    selectedChecksum = "unknown";
+    if((configChecksumList = getenv("XRD_CSLIST"))) {
```

Comment:
Not that there is any significant performance penalty here but this transformation of XRD_CSLIST should really be done only once.